### DISCUSSION TITLE
Laggy Tesla android

### DISCUSSION THREAD

**Discussion by WillE726** [tesla-android/issue-tracker#242] [Q&A] [answered]
I have the Tesla android connected with Wi-Fi to the Tesla but it seems laggy through the browser.  Has anyone else had this problem and is there a fix for it? 
Thanks

**Comment by adamChrystie** [comment, marked as answer]
When android os screen is visible, check pi’s red power light. If it is blinking, pi is not getting enough power.

post pictures of your pi setup. Include where you use pi when it is slow, picture of your case, picture of how you cool the pi cpu and chips. if pi is overheated, it slows down. 

What material is your pi case? Metal is known to weaken pi wifi signal. 

What model and year is your car? Some early model 3 years have very weak interior wifi signal strength & will require an mcu wifi antenna update. I put a Tesla android setup in a pal’s early model 3 and setup was unusable due to lag if pi was under center armrest but good if closer to infotainment screen and on top of center storage areas. His car will need mcu wifi antenna update. This tutorial will end up on website I hear.

**Reply by WillE726** [reply]
Ok I will post pictures later.  I am using the argon one case and have the pi plugged into a 20watt usb c which might be the problem and need an outlet interior.  I have 2019 Tesla model x.  Thanks

**Reply by adamChrystie** [reply]
You might want to get in touch with the author of this post within Tesla android’s reddit community. They have a “legacy” model X, no sure what year that is and might have tips for you.

https://www.reddit.com/r/teslaandroid/comments/11kt06w/thread_for_mcu2_sx_users_with_issues/?utm_source=share&utm_medium=ios_app&utm_name=ioscss&utm_content=2&utm_term=1

To help diagnose, research to see where your car’s Wi-Fi antenna is located and try to position pi close to that area. Try taking lid off pi. Lag can be due to pi not being able to chat quickly with tesla Wi-Fi…pi over heating.

There is mention of a upcoming tutorial which will appear on tesla android website when next release happens  about improving interior Wi-Fi ability on older model 3 via easy mod to tesla Wi-Fi antenna might apply to older X.

**Reply by mikegapinski** [reply]
Argon One needs to go, it is made from metal and blocks the signal. The antena mod is hard to do in S/X - it requires taking the MCU out. Worth doing only if some other service is done to it. 

Once you get yourself a plastic/open case with a decent cooler just make sure your pi is located as close to the antena as possible(right rear view mirror).

There might be an easier way improve the Wi-Fi reception in those legacy s/x cars without tearing down the dash, there is a harness where the external antenna connects to under a easy to remove piece of trim, it can be disconnected and terminated with a normal antena inside of the car. Not validated for performance yet; I expect you'd be happy with a normal case and keeping the pi as close to the a pillar on the right as possible.

**Reply by WillE726** [reply]
Ok thanks.  Do you have any suggestions on a plastic case?

**Reply by WillE726** [reply]
Or would an acrylic case work ok?

**Reply by mikegapinski** [reply]
Anything but metal. I used acrylic, abs and abs like resin.

**Comment by mikegapinski** [comment]
Resolved - metal blocking wifi signal
